src/structures/idt.rs

```diff
@@ -890,10 +890,204 @@ bitflags! {
     }
 }
 
+#[macro_export]
+/// Set a general handler in an [`InterruptDescriptorTable`].
+/// ```
+/// #![feature(abi_x86_interrupt)]
+/// use x86_64::set_general_handler;
+/// use x86_64::structures::idt::{InterruptDescriptorTable, InterruptStackFrame};
+///
+/// let mut idt = InterruptDescriptorTable::new();
+/// fn my_general_handler(
+///    stack_frame: InterruptStackFrame,
+///    index: u8,
+///    error_code: Option<u64>,
+/// ) {
+///     todo!("handle irq {}", index)
+/// }
+///
+/// // set only one entry
+/// set_general_handler!(&mut idt, my_general_handler, 14);
+///
+/// // set a range of entries
+/// set_general_handler!(&mut idt, my_general_handler, 32..64);
+///
+/// // set all entries
+/// set_general_handler!(&mut idt, my_general_handler);
+/// ```
+macro_rules! set_general_handler {
+    ($idt:expr, $handler:ident) => {
+        $crate::set_general_handler!($idt, $handler, 0..=255);
+    };
+    ($idt:expr, $handler:ident, $idx:literal) => {
+        $crate::set_general_handler!($idt, $handler, $idx..=$idx);
+    };
+    ($idt:expr, $handler:ident, $range:expr) => {{
+        fn set_general_handler(
+            idt: &mut $crate::structures::idt::InterruptDescriptorTable,
+            range: impl core::ops::RangeBounds<usize>,
```

Comment:
Currently `InterruptDescriptorTable` only uses `usize` for indexing operations and doesn't allow indexing with `u8`, so I tried to mirror that api. I agree that using `u8` would make more sense. Perhaps we should also implement `Index<u8>` and `IndexMut<u8>` for `InterruptDescriptorTable` in a seperate pr.

Reply:
The problem with implementing both `Index<u8>` and `Index<usize>` might be that type annotations are required in more cases. For example, `idt[20]` leads to the non-helpful error message "`idt::InterruptDescriptorTable` cannot be indexed by `i32`" because integer literals default to `i32` in these cases.

However, I'm happy to change the `Index` impl from `usize` to `u8` in the next breaking release. We should do this separately though, since this PR is a non-breaking change otherwise. We can of course already use `u8` for the macro if we want.